trio/_core/_run.py

```diff
@@ -2358,7 +2357,9 @@ def my_done_callback(run_outcome):
     # spawn_system_task. We don't actually run any user code during
     # this time, so it shouldn't be possible to get an exception here,
     # except for a TrioInternalError.
-    next_send = None
+    next_send: EventResult = cast(
+        EventResult, None
+    )  # First iteration must be `None`, every iteration after that is EventResult
```

Comment:
do we need really need to explicitly type this?

Reply:
Casting None to EventResult yes, but you are probably right about not having to explicitly type it after the cast.

Reply:
Changed in dacd4b3